lib/linux_admin/ssh_agent.rb

```diff
@@ -0,0 +1,85 @@
+require 'net/ssh'
+require 'fileutils'
+module LinuxAdmin
+  class SshAgent
+    attr_reader :pid
+    attr_reader :sock
+
+    def initialize(agent_socket, ssh_private_key)
+      FileUtils.mkdir_p(File.dirname(agent_socket))
+      agent_details = `ssh-agent -a #{agent_socket}`
+      @sock         = agent_details.split("=")[1].split(" ")[0].chop
+      @pid          = agent_details.split("=")[2].split(" ")[0].chop
+      IO.popen({"SSH_AUTH_SOCK" => @sock, "SSH_AGENT_PID" => @pid}, ["ssh-add", "-"], :mode => 'w') do |f|
+        f.puts(ssh_private_key)
+        if $?.to_i != 0
```

Comment:
It doesn't seem like `IO.popen` sets `$?` until after the block, so this should be moved to the outside.
From http://ruby-doc.org/core-2.2.4/IO.html#method-c-popen:
>At the end of block, Ruby closes the pipe and sets $?. In this case IO.popen returns the value of the block.